reV/supply_curve/supply_curve.py

```diff
@@ -634,26 +673,97 @@ def full_sort(self, trans_table=None, sort_on='total_lcoe',
 
-        if connected != len(self):
+        unconnected = np.where(self._mask[self._sc_gids])[0].tolist()
+        if unconnected:
             msg = ("{} supply curve points were not connected to tranmission! "
                    "Unconnected sc_gid's: {}"
-                   .format(len(self) - connected,
-                           np.where(self._mask[self._sc_gids])[0].tolist()))
+                   .format(len(unconnected), unconnected))
             logger.warning(msg)
             warn(msg)
 
-        return connections
+        supply_curve = self._sc_points.merge(connections, on='sc_gid')
+        if comp_wind_dirs is not None:
+            sc_gids = comp_wind_dirs.sc_gids
+            mask = supply_curve['sc_gid'].isin(sc_gids)
+            supply_curve = supply_curve.loc[mask]
+
+        return supply_curve
+
+    def full_sort(self, trans_table=None, sort_on='total_lcoe',
+                  columns=('trans_gid', 'trans_capacity', 'trans_type',
+                           'trans_cap_cost', 'dist_mi', 'lcot', 'total_lcoe'),
+                  wind_dirs=None, n_dirs=2, downwind=False):
+        """
+        run supply curve sorting in serial
+
+        Parameters
+        ----------
+        trans_table : pandas.DataFrame | NoneType
+            Supply Curve Tranmission table to sort on
+            If none use self._trans_table
+        sort_on : str
+            Column label to sort the Supply Curve table on. This affects the
+            build priority - connections with the lowest value in this column
+            will be built first.
+        columns : list | tuple
+            Columns to preserve in output connections dataframe.
+        wind_dirs : pandas.DataFrame | str
+            path to .csv or reVX.wind_dirs.wind_dirs.WindDirs output with
+            the neighboring supply curve point gids and power-rose value at
+            each cardinal direction
+        n_dirs : int, optional
+            Number of prominent directions to use, by default 2
+        downwind : bool, optional
+            Flag to remove downwind neighbors as well as upwind neighbors,
+            by default False
+
+        Returns
+        -------
+        supply_curve : pandas.DataFrame
+            Updated sc_points table with transmission connections, LCOT
+            and LCOE+LCOT based on full supply curve connections
+        """
+        if trans_table is None:
+            trans_table = self._trans_table
+
+        if isinstance(columns, tuple):
+            columns = list(columns)
+
+        pos = trans_table['lcot'].isnull()
+        trans_table = trans_table.loc[~pos].sort_values(sort_on)
+
+        total_lcoe_fric = None
+        if self._consider_friction and 'mean_lcoe_friction' in trans_table:
+            columns.append('total_lcoe_friction')
+            total_lcoe_fric = trans_table['total_lcoe_friction'].values
+
+        if wind_dirs is not None:
+            comp_wind_dirs = CompetitiveWindFarms(wind_dirs, self._sc_points,
```

Comment:
I think comp_wind_dirs should be set to None above the if statement and then have it be a kwarg for the protected method _full_sort

Reply:
yup, see above